src/main.rs

```diff
@@ -1,51 +1,78 @@
-use std::io::{Write, stdout};
+use std::{io::{Write, stdout}, fs, collections::HashMap};
 use clap::Parser;
 use crate::game::Game;
 
 pub mod game;
 
-#[derive(Parser,Debug)]
+#[derive(Parser, Debug)]
 struct Args {
     #[arg(short, default_value_t = 6)]
-    tries:u32,
+    tries: u32,
 
     #[arg(short, default_value_t = 5)]
-    word_length:usize,
+    word_length: usize,
+
+    #[arg(short,long, default_value = "english")]
+    language: String,
 }
 
 fn clear(){
     println!("{}[2J", 27 as char);
 }
 
+fn load_languages() -> HashMap<String, HashMap<String,String>>{
+    let mut lang_map:HashMap<String, HashMap<String,String>> = HashMap::new();
+    let langs = fs::read_dir("./languages").unwrap();
+    for lang in langs{
+        let file_name = lang.unwrap().file_name();
+        let lang_name = file_name.to_str().unwrap().trim_end_matches(".txt").to_string();
+        let mut map:HashMap<String,String> = HashMap::new();
+        let file = fs::read_to_string(format!("./languages/{}",file_name.to_str().unwrap()));
+        for line in file.unwrap().lines() {
+            let key = line.split(":").next().unwrap().to_string();
+            let value = line.split(":").nth(1).unwrap().to_string();
+            map.insert(key, value);
+        }
+        lang_map.insert(lang_name, map);
+    }
+    lang_map
+}
+
 fn main() {
     let args = Args::parse();
-    let mut game = Game::new(args.word_length,args.tries);
+    let mut game = Game::new(args.word_length,args.tries,args.language.clone());
```

Comment:
Hvis Game::new tog en &str i stedet for String, ville du undgå clone og kunne bruge &String -> &str type coercion.